fe/fe-core/src/main/java/com/starrocks/catalog/Database.java

```diff
@@ -117,12 +117,64 @@ public Database(long id, String name) {
             this.fullQualifiedName = "";
         }
         this.rwLock = new QueryableReentrantReadWriteLock(true);
-        this.idToTable = new ConcurrentHashMap<>();
-        this.nameToTable = new ConcurrentHashMap<>();
+        this.tableStore = new TableStore();
         this.dataQuotaBytes = FeConstants.default_db_data_quota_bytes;
         this.replicaQuotaSize = FeConstants.default_db_replica_quota_size;
     }
 
+    private class TableStore {
+        private Map<String, Table> nameToTable;
+        private Map<Long, Table> idToTable;
+        private Set<String> nameSet;
+        public TableStore() {
+            this.idToTable = new ConcurrentHashMap<>();
+            this.nameToTable = new ConcurrentHashMap<>();
+            this.nameSet = new HashSet<>();
```

Comment:
should use  concurrent set ?

Reply:
There is no concurrent set under the java.util.concurrent package

Reply:
the key point here is that we need to protect this.nameSet because there are concurrent writes to this set

Reply:
the key point here is that we need to protect this.nameSet because there are concurrent writes to this set

Reply:
and we can use `Sets.newConcurrentHashSet()`

Reply:
done. 
btw  currently tableStore write operations are all under lock